testing/run_tests.py

```diff
@@ -218,80 +221,86 @@ def RunCCTests(build_dir, filter, coverage, capture_core_dump):
-
-  RunEngineExecutable(build_dir, 'dart_plugin_registrant_unittests', filter, shuffle_flags, coverage=coverage)
-
-  if not IsWindows():
-    # https://github.com/flutter/flutter/issues/36295
-    RunEngineExecutable(build_dir, 'shell_unittests', filter, shuffle_flags, coverage=coverage)
-    # https://github.com/google/googletest/issues/2490
-    RunEngineExecutable(build_dir, 'android_external_view_embedder_unittests', filter, shuffle_flags, coverage=coverage)
-    RunEngineExecutable(build_dir, 'jni_unittests', filter, shuffle_flags, coverage=coverage)
-    RunEngineExecutable(build_dir, 'platform_view_android_delegate_unittests', filter, shuffle_flags, coverage=coverage)
-
-  # The image release unit test can take a while on slow machines.
-  RunEngineExecutable(build_dir, 'ui_unittests', filter, shuffle_flags + ['--timeout=90'], coverage=coverage)
-
-  RunEngineExecutable(build_dir, 'testing_unittests', filter, shuffle_flags, coverage=coverage)
-
-  # The accessibility library only supports Mac and Windows.
-  if IsMac() or IsWindows():
-    RunEngineExecutable(build_dir, 'accessibility_unittests', filter, shuffle_flags, coverage=coverage)
-
-  # These unit-tests are Objective-C and can only run on Darwin.
-  if IsMac():
-    RunEngineExecutable(build_dir, 'flutter_channels_unittests', filter, shuffle_flags, coverage=coverage)
-    RunEngineExecutable(build_dir, 'flutter_desktop_darwin_unittests', filter, shuffle_flags, coverage=coverage)
-
-  # https://github.com/flutter/flutter/issues/36296
-  if IsLinux():
     icu_flags = ['--icu-data-file-path=%s' % os.path.join(build_dir, 'icudtl.dat')]
-    RunEngineExecutable(build_dir, 'txt_unittests', filter, icu_flags + shuffle_flags, coverage=coverage)
+    unittests += [
+      make_test('flow_unittests', flags=repeat_flags + ['--'] + flow_flags),
+      make_test('flutter_glfw_unittests'),
+      make_test('flutter_linux_unittests', extra_env={'G_DEBUG': 'fatal-criticals'}),
+      # https://github.com/flutter/flutter/issues/36296
+      make_test('txt_unittests', flags=repeat_flags + ['--'] + icu_flags),
+    ]
+  else:
+    flow_flags = ['--gtest_filter=-PerformanceOverlayLayer.Gold']
+    unittests += [
+      make_test('flow_unittests', flags=repeat_flags + flow_flags),
+    ]
 
-  if IsLinux():
-    gtk_flags = ['--icu-data-file-path=%s' % os.path.join(build_dir, 'icudtl.dat')]
-    RunEngineExecutable(build_dir, 'flutter_linux_unittests', filter, shuffle_flags, coverage=coverage,
-                        extra_env={'G_DEBUG': 'fatal-criticals'})
-    RunEngineExecutable(build_dir, 'flutter_glfw_unittests', filter, shuffle_flags, coverage=coverage)
+  for test, flags, extra_env in unittests:
+    RunEngineExecutable(build_dir, test, filter, flags, coverage=coverage,
+                        extra_env=extra_env, gtest=True)
 
-  # Impeller tests are only supported on macOS for now.
   if IsMac():
+    # flutter_desktop_darwin_unittests uses global state that isn't handled
```

Comment:
I created an issue: https://github.com/flutter/flutter/issues/104789 to track a potential work to make the tests avoid using global states. Maybe we can add a reference to the issue here?

Reply:
Done.